core/trino-main/src/test/java/io/trino/execution/buffer/BenchmarkBlockSerde.java

```diff
@@ -81,7 +81,7 @@ public class BenchmarkBlockSerde
 {
     private static final DecimalType LONG_DECIMAL_TYPE = createDecimalType(30, 5);
 
-    public static final int ROWS = 10_000_000;
+    public static final int ROWS = 8192;
```

Comment:
I don't follow why this is better.
The actual size of pages produced in benchmark setup is decided by `io.trino.spi.PageBuilder#isFull` and that is unchanged. But we have now reduced the total number of rows generated as input for the benchmark, I think this will cut input of the benchmark to one 8K Page. Wouldn't we want to run the JMH for at least a few dozen pages ?

Reply:
Anything that happens in the setup of a JMH benchmark does not count towards the benchmarked method's JMH profile result. As for the profile of the JMH benchmark itself, you should ignore everything coming from the setup method in that.
You can change `io.trino.execution.buffer.BenchmarkDataGenerator#createValues` to provide values iteratively instead of allocating a large array.
I think cutting the size of the input to the benchmark to make the setup shorter makes the JMH result less realistic.